test/e2e/main_test.go

```diff
@@ -29,148 +30,116 @@ import (
-
-const (
-	// LabelSizeSmall is used for tests that usually complete in a minute.
-	LabelSizeSmall = "small"
-
-	// LabelSizeLarge is used for test that usually complete in over a minute.
-	LabelSizeLarge = "large"
-)
-
+// TODO(phisco): make it configurable
 const namespace = "crossplane-system"
 
+// TODO(phisco): make it configurable
 const crdsDir = "cluster/crds"
 
-// The caller (e.g. make e2e) must ensure these exists.
+// The caller (e.g. make e2e) must ensure these exist.
 // Run `make build e2e-tag-images` to produce them
 const (
+	// TODO(phisco): make it configurable
 	imgcore = "crossplane-e2e/crossplane:latest"
-	imgxfn  = "crossplane-e2e/xfn:latest"
 )
 
 const (
-	helmChartDir    = "cluster/charts/crossplane"
+	// TODO(phisco): make it configurable
+	helmChartDir = "cluster/charts/crossplane"
+	// TODO(phisco): make it configurable
 	helmReleaseName = "crossplane"
 )
 
-// FieldManager is the server-side apply field manager used when applying
-// manifests.
-const FieldManager = "crossplane-e2e-tests"
-
-// HelmOptions valid for installing and upgrading the Crossplane Helm chart.
-// Used to install Crossplane before any test starts, but some tests also use
-// these options - for example to reinstall Crossplane with a feature flag
-// enabled.
-func HelmOptions(extra ...helm.Option) []helm.Option {
-	o := []helm.Option{
-		helm.WithName(helmReleaseName),
-		helm.WithNamespace(namespace),
-		helm.WithChart(helmChartDir),
-		// wait for the deployment to be ready for up to 5 minutes before returning
-		helm.WithWait(),
-		helm.WithTimeout("5m"),
-		helm.WithArgs(
-			// Run with debug logging to ensure all log statements are run.
-			"--set args={--debug}",
-			"--set image.repository="+strings.Split(imgcore, ":")[0],
-			"--set image.tag="+strings.Split(imgcore, ":")[1],
-
-			"--set xfn.args={--debug}",
-			"--set xfn.image.repository="+strings.Split(imgxfn, ":")[0],
-			"--set xfn.image.tag="+strings.Split(imgxfn, ":")[1],
-		),
-	}
-	return append(o, extra...)
-}
+var E2EConfig = config.NewFromFlags()
 
 var (
 	// The test environment, shared by all E2E test functions.
 	environment env.Environment
 	clusterName string
 )
 
+func init() {
+	// Set the default suite, to be used as base for all the other suites.
+	E2EConfig.AddDefaultTestSuite(
+		config.WithoutBaseDefaultTestSuite(),
+		config.WithHelmInstallOpts(
+			helm.WithName(helmReleaseName),
+			helm.WithNamespace(namespace),
+			helm.WithChart(helmChartDir),
+			// wait for the deployment to be ready for up to 5 minutes before returning
+			helm.WithWait(),
+			helm.WithTimeout("5m"),
+			helm.WithArgs(
+				// Run with debug logging to ensure all log statements are run.
+				"--set args={--debug}",
+				"--set image.repository="+strings.Split(imgcore, ":")[0],
+				"--set image.tag="+strings.Split(imgcore, ":")[1],
+			),
+		),
+		config.WithLabelsToSelect(features.Labels{
+			config.LabelTestSuite: []string{config.TestSuiteDefault},
+		}),
+	)
+}
```

Comment:
sure, I was keeping it in the init to have it symmetrical with the other files, but doesn't change much. Will do.

Reply:
FWIW I can see the point of keeping it symmetrical - I don't feel strongly here.

I generally am a little averse to using `init`, but that's mostly because I try to avoid package-scoped state. That feels a little less important in tests, and besides it's unavoidable with e2e-framework given the `environment` state.